cuda/insert.cpp

```diff
@@ -0,0 +1,39 @@
+#include <bits/stdc++.h>
+#include <cuda_runtime.h>
+
+#include "kernels.h"
+
+using namespace std;
+
+void insert_point(Point new_point, Grid *root_grid,
+				  vector<QuadrantBoundary> &boundaries,
+				  unordered_map<int, Grid *> &grid_map, int quadrant_id) {
+	// Access the target grid from the unordered map
+	Grid *target_grid = grid_map[quadrant_id];
```

Comment:
since we already found the target grid, can we just pass that here? It will reduce a call to map.

Reply:
I could do that but this is an unordered_map and we incur no overhead of finding something within it since find is O(1) operation

Reply:
Yeah but we're not actually using grid_map anywhere else in the function. So it doesn't make sense. Your call.